consoles/sshVirtsh.pm

```diff
@@ -339,6 +339,7 @@ sub _copy_image_vmware ($self, $name, $backingfile, $file_basename, $vmware_open
       'done;' .
       'else ' .
       "cp /vmfs/volumes/$vmware_nfs_datastore/$nfs_dir/$file_basename $vmware_openqa_datastore;" .
+      "[ $? -ne 0 ] && cp /vmfs/volumes/$vmware_nfs_datastore/$nfs_dir/fixed/$file_basename $vmware_openqa_datastore;" .
```

Comment:
No code so far in os-autoinst needs to do anything with the "fixed" directory which is only something that openQA knows. I would like to keep this design this separated. As also your git commit message does not state the reason or what problem this addresses this shouldn't be accepted as-is.

Reply:
Sorry, it is not clear to me, but `fixed` is a subdirectory of `hdd/` (directory that is included into the os-autinst design), but do not managed (forgotten ?) into that routine: by design _the image is copied from `hdd`, but in this case it is present in the `hdd/fixed` subfolder instead_.
The commit message may be is too short,but it could be updated.
But also, in the PR header I explained the problem occurred, unless explanation was not clear yet: should I better explain that error?

Reply:
> No code so far in os-autoinst needs to do anything with the "fixed" directory which is only something that openQA knows.

Generally I would agree with you Oli , this is implementation details which should be hidden in `os-autoinst` and test code should not be aware about existence of `fixed`  folder. But also I must say that it is "too late" for such comment :man_shrugging: . Major part of things what happen in  `_copy_image_vmware` should happen in os-autoinst and should be hidden from test code . So there are two options how to proceed further : 
- accept the fact that from now on test code will know about `fixed` folder and increase even more tightly coupling of os-autoinst and `osada`
- move fully or partially logic from `osada` into os-autoinst so test code will be able transparently copy image no matter if it is in `fixed` folder or just in `hdd` folder 

P.S. Please bare with me , my knowledge of this piece of code is result of 5 minutes reading of this PR and comments to it so I might missing some obvious things here . So feel free fully or partially ignore it :smile:

Reply:
> Sorry, it is not clear to me, but `fixed` is a subdirectory of `hdd/`, directory included into the os-autinst design, but that subfolder is not managed into that routine: by design _the image is copied from `hdd`, but in this case it is present in the `hdd/fixed` subfolder instead_.

A simple `git grep` reveals that "fixed/" is not used anywhere in os-autoinst so far and hence also not used in any other backend. That's why I wonder what could be the right solution to not hard-code the fixed folder path here.

> The commit message may be is too short,but it could be updated. But also, in the PR header I explained the problem occurred, unless explanation was not clear yet: should I better explain that error and needed changes?

What you put into the description of the pull request could also be in the commit message details. I don't know how you created your pull request but your git commit messages has *only* the subject line while the pull request description has more details. Please keep in mind that the github pull request description is only visible on github, the commit can be considered permanent information storage.

I can recommend the tool [hub](https://github.com/github/hub) (zypper in rubygem-hub) for easier PR creation. Also, I myself use a script [git-pr-last](http://github.com/okurz/scripts/tree/master/git-pr-last) to create a PR with proper description for these simple one-commit PRs:
```bash
$ cat $(which git-pr-last )
#!/bin/sh -e
target="${target:-"$USER"}"
git push $target && git show --no-patch --format=%B | hub pull-request -F -
```

You might want to use the more up-to-date tool https://github.com/cli/cli

Reply:
Actual problem is that other than `hdd/` folder, its subfolder (`fixed`) too is actually popolated by some images to test, somewhere not managed yet so this misalignment may cause errors.
So, alike in the last poo's note written, clear rules for files in this last folder should be properly defined, documented and every appropriate code updated.

Reply:
Right. So how about a `find` call to recurse into any subfolder?

Reply:
Restarting the activities I did an attempt to find a desired image under hdd/ and subfolders: that process consumed resources in cpu and took long time to provide result, being many images present, to skip.
Moreover we already know the original file position, provided by [search_image_on_svirt_host](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/800edc414310ac6bafc982c7221eff4589c78137/tests/installation/bootloader_svirt.pm#L46) hence not needed that `find`.